src/Datadog.Trace/Vendors/Serilog/Core/Logger.cs

```diff
@@ -33,7 +33,7 @@ namespace Datadog.Trace.Vendors.Serilog.Core
     /// be disposed to flush any events buffered within it. Most application
     /// code should depend on <see cref="ILogger"/>, not this class.
     /// </summary>
-    internal sealed class Logger : ILogger, ILogEventSink, IDisposable
+    internal sealed class Logger : ILogger, ILogEventSink, IDisposable, ICoreLogger
```

Comment:
Adding ICoreLogger here lets us pull this into Core as a strategy

Reply:
Every time we modify vendor code it makes me wonder if we're doing this right. Did we consider, for example, writing a light wrapper around `Logger` that implements `ICoreLogger`?

Reply:
I had weighed doing that (with the extra class and needing to implement the constructor passthroughs) and opted for this strategy as changing less and more lightweight.
If you feel strongly either way, I'm okay with creating a wrapper.